lib/avromatic/model/raw_serialization.rb

```diff
@@ -29,6 +29,9 @@ def recursive_serialize(value, name: nil, member_types: nil)
             if value.is_a?(Avromatic::Model::Attributes)
               hash = value.value_attributes_for_avro
               if Avromatic.use_custom_datum_writer
+                unless value.class.config.mutable
+                  hash = { Avromatic::IO::ENCODING_PROVIDER => value }
```

Comment:
That's the approach I wanted to take but it was complicated by the way we serialize union members by stashing an index into the hash. In order to avoid wrapping value in a hash we'd need to figure out some other way to plumb the union member index. I think its doable but I'm not convinced it would be any prettier...